### PR TITLE
Pystray

### PR DESCRIPTION
**Pull request by sphh**
This is my take on a system tray, if you want to include it.

I also made some little tweaks to the configuration page. Maybe you like it.

### CODE REVIEW

File: scriptExamples/extendedMode/system-tray/scriptable_systemtrayicon.py [hunk 1]
```diff
@@ -0,0 +1,765 @@
...
+}
+
+
+# The logging system
+if CONFIG['logging']:
+    import logging
+    logging.basicConfig(
+        level=getattr(logging, CONFIG['logging']),
+        format='%(levelname)8s %(funcName)s%(message)s',
+        filename=f'/home/sph/{pathlib.Path(__file__).with_suffix(".log").name}')
```
Comment (electrotype) [inline scriptExamples/extendedMode/system-tray/scriptable_systemtrayicon.py:191]: Shouldn't this path be a variable?

**Review by electrotype — COMMENTED** [review]
Nice script example!

Can you tell what are the requirements for it (for the doc)? I see that at least Pything 3.6 is required. 

Is it supposed to work on Windows too? If so, I guess you should be even more careful about the paths (see my other comment).

Thanks!

File: scriptExamples/extendedMode/system-tray/scriptable_systemtrayicon.py (hunk 1 above)
Comment (sphh) [reply scriptExamples/extendedMode/system-tray/scriptable_systemtrayicon.py:191]: Oops, you are absolutely right!

Easy to change, but do we need logging at all?

**Comment by sphh** [comment]
There is some documentation in the doc-string, which also mentions under "Installation / This script", which packages are needed (`pystray` and `Pillow`). Should this be under a separate heading?

File: scriptExamples/extendedMode/system-tray/scriptable_systemtrayicon.py (hunk 1 above)
Comment (electrotype) [reply scriptExamples/extendedMode/system-tray/scriptable_systemtrayicon.py:191]: To be honest, you decide what's in your script. So as you wish!

**Comment by electrotype** [comment]
> There is some documentation in the doc-string, which also mentions under "Installation / This script", which packages are needed (`pystray` and `Pillow`). Should this be under a separate heading?

So `pystray` and `Pillow`, perfect. 

Should it work on Windows too?

File: scriptExamples/extendedMode/system-tray/scriptable_systemtrayicon.py (hunk 1 above)
Comment (sphh) [reply scriptExamples/extendedMode/system-tray/scriptable_systemtrayicon.py]: Done.

**Comment by sphh** [comment]
Re `pystray`: As long as `pystray` works, yes. And according to the `pystray` website, it should work on Linux, Windows and MacOS.

**Comment by electrotype** [comment]
Merged and released in version 5.1.

Note that I fixed a bug that prevented the "Options Saved!" message to appear when using "simple" mode: https://github.com/electrotype/thunderbird-addon-scriptable-notifications/commit/c6a5ff46dcef45869764c73e7b17afb3651198c9#diff-16f022ec89282b37542e3374910a5e4a4cf37596abf8f723f0d7f8ab1096f853R84 . 

Feel free to test with your extended/connection-based scripts to validate I didn't break anything (should not).

**Comment by sphh** [comment]
I tried the newest commit and I get the "Options Saved!" message with extended and connection based mode.

Your version of the tip is much better than mine. Thanks!

**Pull request merged**